internal/daemon/api_plan.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"gopkg.in/yaml.v3"
+)
+
+func v1GetPlan(c *Command, r *http.Request, _ *userState) Response {
+	format := r.URL.Query().Get("format")
+	if format != "yaml" {
+		return statusBadRequest("invalid format %q", format)
+	}
+
+	servmgr := c.d.overlord.ServiceManager()
+	plan, err := servmgr.Plan()
+	if err != nil {
+		return statusInternalError("cannot get plan: %v", err)
+	}
+	planYAML, err := yaml.Marshal(plan)
+	if err != nil {
+		return statusInternalError("cannot serialize plan: %v", err)
```

Comment:
This is the counter example: the error message here is relevant as it's adding lacking context without which the message would be meaningless in the Pebble context.